packages/flutter_tools/test/commands.shard/hermetic/pub_get_test.dart

```diff
@@ -114,6 +114,22 @@ void main() {
     FileSystem: () => fileSystem,
   });
 
+  testUsingContext('pub get throws error on missing directory', () async {
+    final PackagesGetCommand command = PackagesGetCommand('get', false);
+    final CommandRunner<void> commandRunner = createTestCommandRunner(command);
+
+    try {
+      await commandRunner.run(<String>['get', 'missing_dir']);
+      fail('expected an exception');
+    } on Exception catch (e) {
+      expect(e.toString().contains('Expected to find project root in missing_dir'), true);
```

Comment:
nit, use contains matcher to get a better failure message if/when it fails:

```suggestion
      expect(e.toString(), contains('Expected to find project root in missing_dir'));
```